tests/unit_tests/data_steward/cdr_cleaner/cleaning_rules/replace_standard_id_in_domain_tables_test.py

```diff
@@ -290,11 +258,63 @@ def test_parse_src_concept_id_logging_query(
             domain_concept_id=self.domain_concept_id,
             domain_source=self.domain_source_concept_id)
 
-        actual_query = self.rule_instance.parse_src_concept_id_logging_query(
+        actual_query = self.rule_instance.parse_src_concept_id_logging_subquery(
             self.condition_table)
 
         self.assertCountEqual(expected_query, actual_query)
 
+    @patch.object(ReplaceWithStandardConceptId,
+                  'parse_src_concept_id_logging_subquery')
+    def test_parse_src_concept_id_logging_query(
+        self, mock_parse_src_concept_id_logging_subquery):
+        query_1 = 'LOGGING QUERY condition'
+        query_2 = 'LOGGING QUERY procedure'
+        mock_parse_src_concept_id_logging_subquery.side_effect = [
+            query_1, query_2
+        ]
+        expected_query = UNION_ALL.join([query_1, query_2])
+        actual_query = self.rule_instance.parse_src_concept_id_logging_query()
+        self.assertEqual(expected_query, actual_query)
+
+        self.assertEqual(mock_parse_src_concept_id_logging_subquery.call_count,
+                         2)
+        mock_parse_src_concept_id_logging_subquery.assert_any_call(
+            self.condition_table)
+        mock_parse_src_concept_id_logging_subquery.assert_called_with(
+            self.procedure_table)
+
+    def test_parse_duplicate_id_update_subquery(self):
+        expected_query = DUPLICATE_ID_SUB_QUERY.render(
+            table_name=self.condition_table,
+            project=self.project_id,
+            dataset=self.dataset_id,
+            sandbox_dataset=self.sandbox_id,
+            logging_table=SRC_CONCEPT_ID_TABLE_NAME)
+        actual_query = self.rule_instance.parse_duplicate_id_update_subquery(
+            self.condition_table)
+        self.assertEqual(expected_query, actual_query)
+
+    @patch.object(ReplaceWithStandardConceptId,
+                  'parse_duplicate_id_update_subquery')
+    def test_parse_duplicate_id_update_query(
+        self, mock_parse_duplicate_id_update_subquery):
+        query_1 = 'UPDATE LOGGING QUERY condition'
+        query_2 = 'UPDATE LOGGING QUERY procedure'
+        mock_parse_duplicate_id_update_subquery.side_effect = [query_1, query_2]
+        expected_query = DUPLICATE_ID_UPDATE_QUERY.render(
+            project=self.project_id,
+            sandbox_dataset=self.sandbox_id,
+            logging_table=SRC_CONCEPT_ID_TABLE_NAME,
+            unioned_query=UNION_ALL.join([query_1, query_2]))
+        actual_query = self.rule_instance.parse_duplicate_id_update_query()
+        self.assertEqual(expected_query, actual_query)
+
+        self.assertEqual(mock_parse_duplicate_id_update_subquery.call_count, 2)
+        mock_parse_duplicate_id_update_subquery.assert_any_call(
+            self.condition_table)
+        mock_parse_duplicate_id_update_subquery.assert_called_with(
+            self.procedure_table)
```

Comment:
I recommend we remove these tests. At last eng sync we agreed we should eliminate SQL generation stuff from unit tests. The most valuable way to test our SQL will be through integration tests. Sticking to a test-per-method does not usually do us any good and these kinds of tests tend to be circular and change-prone. Your integration tests look good to me. That you were able to implement this optimization without modifying them reinforces that.

Reply:
This is done